### PR TITLE
[Wallet] Kill Accounts

### LINKED ISSUE #1706: [Tracking] Remove wallet 'account' API
This is a tracking issue for the removal of the wallet's built-in 'account' API, which is clunky, error-prone, and already marked deprecated. An earlier issue (#1248) outlined the general timeline plan for this removal, this issue will track actionable changes required for each stage.

Stage 1 (v4.0.1 Jan 13 2020, *completed*):
 - [x] [RPC] Notate all account stuff as deprecated (#1251)

Stage 2 (v4.2.0 TBA, *completed* #1731):
 - [x] Rename account to label where appropriate (ref: https://github.com/bitcoin/bitcoin/pull/11536)
 - [x] Simplify getbalance implementation (ref: https://github.com/bitcoin/bitcoin/pull/9614)
 - [x] Give RPC commands more information about the RPC request (ref: https://github.com/bitcoin/bitcoin/pull/8788)
 - [x] Introduce 'label' API for wallet (ref: https://github.com/bitcoin/bitcoin/pull/12892)
 - [x] Disable account RPC functions by default. Wallet functional tests updated to use -deprecatedrpc=accounts to maintain existing behaviour (ref: https://github.com/bitcoin/bitcoin/pull/12953)
 - [x] Remove `getlabeladdress` intermediary RPC method (ref: https://github.com/bitcoin/bitcoin/pull/13060)
 - [x] Add filter option to listreceivedbyaddress (ref: https://github.com/bitcoin/bitcoin/pull/9991)
 - [x] Remove 'account' API from wallet functional tests (ref: https://github.com/bitcoin/bitcoin/pull/13075)
 - [x] Fixups from account API deprecation (ref: https://github.com/bitcoin/bitcoin/pull/13498)

Stage 3 (v5.0.0 TBA, *pending*):
 - [ ] Fully remove the 'account' API (ref: https://github.com/bitcoin/bitcoin/pull/13825 = https://github.com/bitcoin/bitcoin/commit/c9c32e6b844fc79467b7e24c6c916142a0d08484)

### PR DESCRIPTION
**Pull request by Fuzzbawls**
This finally removes the last remnants of the account API, which was
deprecated in v4.2.0 in favor of a new label API.

Up until this commit, it was possible to still use the old account API
by starting the wallet with `-deprecatedrpc=account`, but this
functionality is now removed.

Closes #1706

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2394,13 +1949,9 @@ UniValue listtransactions(const JSONRPCRequest& request)
...
             "    \"address\":\"pivxaddress\",    (string) The pivx address of the transaction. Not present for \n"
             "                                                move transactions (category = move).\n"
-            "    \"category\":\"send|receive|move\", (string) The transaction category. 'move' is a local (off blockchain)\n"
-            "                                                transaction between accounts, and not associated with an address,\n"
-            "                                                transaction id or block. 'send' and 'receive' transactions are \n"
-            "                                                associated with an address, transaction id and block details\n"
-            "    \"amount\": x.xxx,          (numeric) The amount in PIV. This is negative for the 'send' category, and for the\n"
-            "                                         'move' category for moves outbound. It is positive for the 'receive' category,\n"
-            "                                         and for the 'move' category for inbound funds.\n"
+            "    \"category\":\"send|receive\",  (string) The transaction category.\n"
```
Comment (random-zebra) [inline src/wallet/rpcwallet.cpp:1952]: There are actually other possible categories (`orphan`, `immature`, `generate`).
We, also, probably want to remove the mention to the `move` category in the previous field (address), now that "accounting entries" are a thing of the past.

**Review by random-zebra — APPROVED** [review]
utACK c6634c02b88cb6d35be3d21e83edd7dc8e5db33b

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -2326,116 +1901,43 @@ void ListTransactions(const CWalletTx& wtx, const std::string& strAccount, int n
...
-                entry.pushKV("vout", r.vout);
-                if (fLong)
-                    WalletTxToJSON(wtx, entry);
-                ret.push_back(entry);
+                label = pwalletMain->GetNameForAddressBookEntry(r.destination);
+            UniValue entry(UniValue::VOBJ);
+            if (involvesWatchonly || (::IsMine(*pwalletMain, r.destination) & ISMINE_WATCH_ONLY))
+                entry.pushKV("involvesWatchonly", true);
+            MaybePushAddress(entry, r.destination);
+            if (wtx.IsCoinBase()) {
```
Comment (furszy) [inline src/wallet/rpcwallet.cpp:1911]: Not for this PR but this should be `if (wtx.IsCoinBase() || wtx.IsCoinstake()) {`

**Review by furszy — APPROVED** [review]
good cleanup ☕ , utACK c6634c0 .
Left a comment for a future PR.

**Pull request merged**